xwiki-platform-core/xwiki-platform-wysiwyg/xwiki-platform-wysiwyg-client/src/main/java/org/xwiki/gwt/wysiwyg/client/gadget/SelectGadgetWizardStep.java

```diff
@@ -73,7 +73,7 @@ public Object getResult()
         GadgetInstance gadgetInstance = new GadgetInstance();
         gadgetInstance.setMacroCall(superResult);
         // prefill the title of the gadget with the name of the macro
-        gadgetInstance.setTitle(superResult.getName());
+        gadgetInstance.setTitle("$services.localization.rendered('rendering.macro." + superResult.getName() + "')");
```

Comment:
Building the WYSIWYG editor takes time but you can test in a wiki page:

    $services.localization.rendered('rendering.macro." + superResult.getName() + "')

to see if it works on not :)